external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaTestUtils.scala

```diff
@@ -131,11 +132,33 @@ class KafkaTestUtils(
   }
 
   private def setUpMiniKdc(): Unit = {
-    val kdcDir = Utils.createTempDir()
     val kdcConf = MiniKdc.createConf()
     kdcConf.setProperty(MiniKdc.DEBUG, "true")
-    kdc = new MiniKdc(kdcConf, kdcDir)
-    kdc.start()
+    var bindException = false
+    var kdcDir: File = null
+    var numRetries = 1
+    do {
```

Comment:
Could you put some notes (e.g., the hadoop jira ID) here about why we need this logic for `MiniKdc`?